### ISSUE TITLE
DB connection fails with port 5432 with docker

### ISSUE THREAD

**Issue by MMCMA** [firewalld/firewalld#1337]
### What happened

I added firewalld to my ubuntu enviroment and experience an issue with connecting to an external DB. 

The DB connection should be established inside a docker container. I added TCP 5432 as an exception to the docker zone but it still fails to connect with the following error.

```
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) connection to server at "XXX.X.XX.XX", port 5432 failed: No route to host
    Is the server running on that host and accepting TCP/IP connections?
```

Once I stop firewalld, I can connect to the DB without problems.

### What you expected to happen

I expect the docker container to connect to the external DB via port 5432

### Firewalld Version

1.1.1

### Firewalld Backend

FirewallBackend=nftables

### Linux distribution

Ubuntu 22.04.4 LTS

### Linux kernel version

5.15.0-105-generic

**Comment by erig0** [comment]
Is the DB running in the container/pod? Or is the pod/container trying to connect to an external DB? Please describe the topology a bit more.

**Comment by MMCMA** [comment]
The container is trying to connect to an external DB that runs on a different host / server.

**Comment by erig0** [comment]
Hrm. All outbound traffic from the container should be allowed by default.

Do you have any issues with other traffic? e.g. an http request

**Comment by MMCMA** [comment]
These are my settings, maybe something I changed causes the issue.

```
sudo firewall-cmd --list-all --zone=docker
docker
  target: ACCEPT
  icmp-block-inversion: no
  interfaces:
  sources:
  services:
  ports: 5432/tcp
  protocols:
  forward: yes
  masquerade: no
  forward-ports:
  source-ports:
  icmp-blocks:
  rich rules:
```


```
sudo firewall-cmd --list-all
You're performing an operation over default zone ('public'),

public
  target: default
  icmp-block-inversion: no
  interfaces:
  sources:
  services: dhcpv6-client ssh
  ports: 22/tcp 5432/tcp
  protocols:
  forward: yes
  masquerade: no
  forward-ports:
  source-ports:
  icmp-blocks:
  rich rules:

```

```
sudo firewall-cmd --list-all --zone=restricted
restricted (active)
  target: default
  icmp-block-inversion: no
  interfaces:
  sources: IP ranges ...
  services: ftp http
  ports: 8501/tcp 22/tcp 5432/tcp
  protocols:
  forward: no
  masquerade: no
  forward-ports:
  source-ports:
  icmp-blocks:
  rich rules:

```

**Comment by erig0** [comment]
It looks like there is nothing (no sources, no interfaces) assigned to the docker zone.

Did you configure this zone yourself? Or did docker set it up?

**Comment by MMCMA** [comment]
it was set-up by docker, but I added port 5432/tcp.

**Comment by erig0** [comment]
Nothing is assigned to the docker zone. That would explain why the container traffic is not working. Maybe try restarting the docker network or daemon.

**Comment by erig0** [comment]
This is an improvement. At least docker0 is assigned to the docker zone now.

You have a `restricted` zones with redacted `sources`. Do any of those sources match the docker IP addresses?

**Comment by MMCMA** [comment]
good point, it solved it - I removed the internal docker subnet and it is working now. The initial issue was that it wasn't working after installing firewalld and then I assumed I need to add the docker subnet explicity, yet the original problem was that I hadn't run `systemctl restart docker` to fix iptables for docker. Once I restarted docker I forgot to remove the docker subnet from restricted. Thank you, very helpful but should've figured it out myself.
